### PR TITLE
Init module

### PR DESCRIPTION
**Pull request by cactusbento**
Closes: https://github.com/ziglang/zig/issues/15653

This adds the command `zig init-mod` to the zig compiler, creating a very basic module in the cwd.

I was not sure how to handle the exhaustive switch statements, so I mostly copied what already existed since modules are built by consumers.

### CODE REVIEW

**Review by andrewrk — CHANGES_REQUESTED** [review]
Hi @cactusbento, welcome, and thank you for working on this contribution.

Consider that this PR, which is only supposed to add a new "init-mod" template, has caused you to edit files such as link.zig, Plan9.zig, and Compilation.zig. This is a clue that something has gone horribly wrong.

Please do not add another enum tag to OutputMode. I think you will find that to greatly simplify the diff.

File: lib/init-mod/build.zig [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // means any target is allowed, and the default is native. Other options
+    // for restricting supported target set are available.
+    const target = b.standardTargetOptions(.{});
+
+    // Standard optimization options allow the person running `zig build` to select
+    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall. Here we do not
+    // set a preferred release mode, allowing the user to decide how to optimize.
+    const optimize = b.standardOptimizeOption(.{});
+
+    _ = b.addModule("$", .{
```
Comment (andrewrk) [inline lib/init-mod/build.zig:18]: you should use it

**Comment by cactusbento** [comment]
> Hi @cactusbento, welcome, and thank you for working on this contribution.
> 
> Consider that this PR, which is only supposed to add a new "init-mod" template, has caused you to edit files such as link.zig, Plan9.zig, and Compilation.zig. This is a clue that something has gone horribly wrong.
> 
> Please do not add another enum tag to OutputMode. I think you will find that to greatly simplify the diff.

Yeah, I kinda expected that. I wrote this up on a whim only following errors the compiler spits at me. Looking back, `cmdInit` has most of the functionality this would require, but limited to OutputMode. So I'll change this to point to a new function for now `cmdInitMod`, and I'll ponder some new way of doing this.

**Comment by andrewrk** [comment]
Sounds good!

File: lib/init-mod/build.zig (hunk 1 above)
Comment (cactusbento) [reply lib/init-mod/build.zig:18]: Got it, I'll move the module to a `lib.zig` and change `main.zig` to an executable that imports this module.

**Comment by andrewrk** [comment]
Please see https://github.com/ziglang/zig/issues/15653#issuecomment-1650625259

**Comment by cactusbento** [comment]
> Please see [#15653 (comment)](https://github.com/ziglang/zig/issues/15653#issuecomment-1650625259)

I'll close this PR and start a new one with init-lib changed instead.

**Pull request closed without merging**